cmd/oras/internal/display/status/interface.go

```diff
@@ -61,3 +63,24 @@ type CopyHandler interface {
 	PostCopy(ctx context.Context, desc ocispec.Descriptor) error
 	OnMounted(ctx context.Context, desc ocispec.Descriptor) error
 }
+
+// ManifestIndexCreateHandler handles status output for manifest index create command.
+type ManifestIndexCreateHandler interface {
+	OnSourceManifestFetching(source string) error
+	OnSourceManifestFetched(source string) error
```

Comment:
Can we remove `Source` from the naming? Will non-source manifest be fetched?

Reply:
This is due to discard implementation has two OnManifestFetching method. To solve this issue,  an interface of ManifestReferenceFetchHandler is created and both index create and index update handlers contain it.